phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -568,13 +568,18 @@ public Integer call() throws SQLException {
                                 checkIfDDLStatementandMutationState(stmt, state);
                                 MutationState lastState = plan.execute();
                                 state.join(lastState);
+                                // Unfortunately, JDBC uses an int for update count, so we
+                                // just max out at Integer.MAX_VALUE
+                                int lastUpdateCount = (int) Math.min(Integer.MAX_VALUE,
+                                        lastState.getUpdateCount());
                                 if (connection.getAutoCommit()) {
                                     connection.commit();
+                                    if (isAtomicUpsert) {
+                                        lastUpdateCount = connection.getMutationState()
+                                                .getNumUpdatedRowsForAutoCommit();
+                                    }
```

Comment:
If you do the `numRowsIgnored` suggestion I stated above then we don't need to change anything here.

Reply:
I thought more about this. I think we will still need some changes here even with my suggestion.

Reply:
The problem is `connection.commit()` calls `resetState` which resets everything including `numRows`

Reply:
Yes, that was one of the reasons to introduce `numUpdatedRowsForAutoCommit`